### DISCUSSION TITLE
Troubleshooting the useSwitchNetwork Hook from wagmi in WalletConnect with Metamask Mobile

### DISCUSSION THREAD

**Discussion by puneettiwariofficial** [WalletConnect/walletconnect-monorepo#2843] [Developer Support]
I'm currently facing an issue while trying to utilize the useSwitchNetwork hook in my WalletConnect integration with the Metamask mobile app. Despite implementing the hook as expected, it does not seem to switch the network, and I'm not receiving any error messages. I would appreciate any assistance or insights regarding this matter.

Here is the code 

```
  const { error, isLoading, pendingChainId, switchNetwork } =   useSwitchNetwork();
  
    useEffect(() => {
    if(chain != CHAIN_ID){
    switchNetwork?.(CHAIN_ID);
    }
  }, [chain]);
```

**Comment by glitch-txs** [comment]
Switching network actually sets a default chain in the dapp side which will be used when triggering a transaction, since the user is already connected to that chain it won't request any switching on the wallet side.

**Reply by puneettiwari61** [reply]
But user is connected to different chain.
